File: app/src/main/java/com/druger/refresher/alarms/AlarmSetter.java
package com.druger.refresher.alarms;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.druger.refresher.App;
import com.druger.refresher.database.RealmHelper;
import com.druger.refresher.models.ModelTask;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

/**
 * Created by druger on 29.09.2015.
 */
public class AlarmSetter extends BroadcastReceiver {

    @Inject
    AlarmHelper alarmHelper;
    @Inject
    RealmHelper realmHelper;

    @Override
    public void onReceive(Context context, Intent intent) {
        App.getAppComponent().inject(this);

        List<ModelTask> tasks = new ArrayList<>();
        tasks.addAll(realmHelper.getTasksByAnyStatus());

        for (ModelTask task : tasks) {
            if (task.getDate() != 0){
                alarmHelper.setAlarm(task);
            }
        }
    }
}

File: app/src/main/java/com/druger/refresher/alarms/AlarmSetter.kt
package com.druger.refresher.alarms

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.druger.refresher.App
import com.druger.refresher.models.ModelTaskNew
import javax.inject.Inject

/**
* Created by druger on 29.09.2015.
*/
class AlarmSetter : BroadcastReceiver() {

    @Inject
    lateinit var alarmHelper: AlarmHelper

    override fun onReceive(context: Context, intent: Intent) {
        App.getApplicationComponent().inject(this)

        val tasks: MutableList<ModelTaskNew> = mutableListOf()

        for (task in tasks) {
            if (task.reminderDate.toInt() != 0) {
                alarmHelper.setAlarm(task)
            }
        }
    }
}
